src/graphnet/models/gnn/dynedge_kaggle_tito.py

```diff
@@ -0,0 +1,253 @@
+"""Implementation of DynEdge architecture used in.
+
+                    IceCube - Neutrinos in Deep Ice
+Reconstruct the direction of neutrinos from the Universe to the South Pole
+
+Kaggle competition.
+
+Solution by TITO.
+"""
+
+from typing import List, Optional
+
+import torch
+from torch import Tensor, LongTensor
+from torch.nn.modules import TransformerEncoder, TransformerEncoderLayer
+from torch_geometric.data import Data
+from torch_geometric.utils import to_dense_batch
+from torch_scatter import scatter_max, scatter_mean, scatter_min, scatter_sum
+
+from graphnet.models.components.layers import DynTrans
+from graphnet.utilities.config import save_model_config
+from graphnet.models.gnn.gnn import GNN
+from graphnet.models.utils import calculate_xyzt_homophily
+
+GLOBAL_POOLINGS = {
+    "min": scatter_min,
+    "max": scatter_max,
+    "sum": scatter_sum,
+    "mean": scatter_mean,
+}
+
+
+class DynEdgeTITO(GNN):
+    """DynEdge (dynamical edge convolutional) model."""
+
+    @save_model_config
+    def __init__(
+        self,
+        nb_inputs: int,
+        features_subset: slice = slice(0, 4),
+        layer_size_scale: int = 4,
+        global_pooling_schemes: List[str] = ["max"],
+    ):
+        """Construct `DynEdge`.
+
+        Args:
+            nb_inputs: Number of input features on each node.
+            features_subset: The subset of latent features on each node that
+                are used as metric dimensions when performing the k-nearest
+                neighbours clustering. Defaults to [0,1,2].
+            layer_size_scale: Integer that scales the size of hidden layers.
+            global_pooling_schemes: The list global pooling schemes to use.
+                Options are: "min", "max", "mean", and "sum".
+        """
+        # DynEdge layer sizes
+        dynedge_layer_sizes = [(256, 256) for layer in range(layer_size_scale)]
```

Comment:
This seems to define both the latent dimensions and number of layers in your DynTrans convolution, so I think the variable should be named `dyntrans_layer_sizes` instead. Also, if you look at the implementation of DynEdge [here](https://github.com/graphnet-team/graphnet/blob/eaa122a75f0b86e5c9e096e681125178cae76875/src/graphnet/models/gnn/dynedge.py#L32), then we allow for the layer sizes to be given as an input, where the number of layers is inferred based on this input. I think it would be ideal to do something similar here, so one can adjust these hyperparameters. So instead of specifying `layer_size_scale: int = 4`, the user could specify `dyntrans_layer_sizes: Optional[List[Tuple[int, ...]]] = None`, and if the parameter is `None`, it could default to the choice you have written here.

Reply:
- Nice suggestion, that's a better name for the variable.
- That was my first approach, however I got CodeClimate error when performing the PR. I will leave as it is, as I think otherwise everytime anyone pull the code and then try to do a PR, it will give an error. I expect that this may change if the PR gets approved, so maybe it is better to use the original approach (_dyntrans_layer_sizes: Optional[List[Tuple[int, ...]]] = None_). Pending on your answer to proceed.

Reply:
Cool! CodeClimate is a guideline; Most of the time the suggestions are good to follow, but in some instances it's impractical and in those cases we approve the PR anyway. 

Take a look at [DynEdge](https://codeclimate.com/github/graphnet-team/graphnet/src/graphnet/models/gnn/dynedge.py/source#issue-2436ce949c626cc70f6b3b34e042d1a8) in CodeClimate. As it exists in the repo today, there are some flags raised by CodeClimate. Those are OK. So if your implementation has the same flags raised, it's OK.

If you still have the code available that lets the layer sizes be defined by the user, as discussed above, I'd suggest that you add it. If you prefer to proceed as-is, then I think we should create a feature request for this functionality and look at it later. Let me know what you prefer!